cmd/earth/main.go

```diff
@@ -972,6 +975,11 @@ func (app *earthApp) run(ctx context.Context, args []string) int {
 				"Check your git auth settings.\n" +
 					"Did you ssh-add today? Need to configure ~/.earthly/config.yml?\n" +
 					"For more information see https://docs.earthly.dev/guides/auth\n")
+		} else if rpcRegex.MatchString(err.Error()) {
```

Comment:
Wondering if using something like this would be better for unwrapping gRPC errors: https://stackoverflow.com/a/52972944

Reply:
Maybe? All the status are `unknown`, at least that I can seem to trigger? Is there a gRPC error parser to pull out the error message?

We would still need to know that it is a gRPC error in the first place; perhaps then just checking for `rpc error:` would be best, assuming theres a built in way to pull out the error text.

Reply:
Based on the SO link, I think you can do
```
e, ok := status.FromError(err)
if ok {
  ...
}
```
to check if it's gRPC error and then
```
e.Message()
```
to grab the message.

Reply:
Yeah I tried that. Doesn't pick it up as a gRPC message; FWIW I am passing in the error here from `errors.Cause(err)` to ensure its the gRPC error.

```
               if st, ok := status.FromError(errors.Cause(err)); ok {
			app.console.Warnf("YES GRPC:\n")
			app.console.Warnf("Error: %v\n", st.Message())
		} else if st, ok := status.FromError(errors.Cause(err)); !ok {
			app.console.Warnf("NO GRPC:\n")
			app.console.Warnf("Error: %v\n", st.Message())
		} 
```

Ends up printing this on my current test case:

```
NO GRPC:
Error: rpc error: code = Unknown desc = syntax error: line 1:0 no viable alternative at input 'no-artifact:'
```

Reply:
Ah ok. Then I think this looks good.

Reply:
OK. Also tried it with removing the `rpc error:` prefix... still no dice